src/fs/loaders.rs

```diff
@@ -325,7 +330,31 @@ impl ApiLoadType {
                 motion_lib::write_stream(&mut cursor, &motion)?;
                 let vec = cursor.into_inner();
                 Ok((vec.len(), vec))
-            }
+            },
+            ApiLoadType::BgmPropertyPatch => {
+                let patches = if let Some(patches) = ApiLoader::get_bgm_property_patches_for_hash(local.smash_hash()?) {
+                    patches
+                } else {
+                    return Err(ApiLoaderError::Other("No patches found for file in bgm_property patch!".to_string()));
+                };
+
+                let data = ApiLoader::handle_load_base_file(local)?;
+                let mut reader = std::io::Cursor::new(data);
+
+                let mut bgm_property = BgmPropertyFile::read(&mut reader).unwrap();
```

Comment:
About the "open" method, check this usage in [smash-arc for example](https://github.com/jam1garner/smash-arc/blob/master/src/arc_file.rs#L154)